### ISSUE TITLE
Generalized Linear Mixed Model, Random Effect and Three Way Interaction

### ISSUE THREAD

**Issue by Whiteknight2119** [glmmTMB/glmmTMB#1049]
Hello,

I am currently running a Generalized Linear Mixed Model with a Three Way Interaction and Random Effect. I have 3 Fixed Effects and 1 Random Effect.

Fixed: Mgt (Management) - No-Till & Conventional Till (Factor) (3 Levels)

Fixed: Moist_Trt (Moisture Treatment) - Wet/Dry & Constant Moisture (2 Levels)

Fixed: Ex_Trt (Exudate Treatment) - Glucose, Oxalic Acid, and H2O (Control) (3 Levels)

Random: Loc (Location) - Location 1 (L1) & Location 2 (L2) (2 Levels)

Response Variables: Microbial Biomass N, Microbial Biomass C, pH, Nitrate, Ammonium, and Inorganic N.

Code:

```r
Gl_MBN_T1_Day8 <- glmmTMB(MBN ~ Mgt*Moist_Trt*Ex_Trt + (1 | Loc),
                 data = Exp2_Data, family = gaussian)
```

I am currently running into an error:

> Error in fitTMB(TMBStruc) : negative log-likelihood is NaN at starting parameter values
Should I run glmer instead?

I was looking into doing a four way interaction because i did the same experiment at different timepoints. Here's my code:

```r
Gl_MBN_Day8_Combined <- glmmTMB(MBN ~ Mgt*Moist_Trt*Ex_Trt*Experiment + (1 | Loc),
data = Exp2_Data, family = gaussian)
```
Am I overfitting the model? How should I go about this?

**Comment by bbolker** [comment]
My first guess is that you're having the package incompatibility problem described [here](https://stackoverflow.com/questions/77466641/error-in-fittmbtmbstruc-negative-log-likelihood-is-nan-at-starting-parameter); see also [here](https://stackoverflow.com/questions/77481539/error-in-initializeptr-function-cholmod-factor-ldeta-not-provided-by-pack).

If that doesn't help we'll need more information ...

**Comment by Whiteknight2119** [comment]
[Exp1_Water_Temporal.csv](https://github.com/user-attachments/files/15889708/Exp1_Water_Temporal.csv)
[Exp2_Water_Temporal.csv](https://github.com/user-attachments/files/15889712/Exp2_Water_Temporal.csv)
[Gross_N_Min_Exp1_CSV.csv](https://github.com/user-attachments/files/15889713/Gross_N_Min_Exp1_CSV.csv)
[Lambo_Exp1_June.csv](https://github.com/user-attachments/files/15889714/Lambo_Exp1_June.csv)
[Lambo_Exp2_June.csv](https://github.com/user-attachments/files/15889715/Lambo_Exp2_June.csv)
[GitHub Master Worksheet.txt](https://github.com/user-attachments/files/15889755/GitHub.Master.Worksheet.txt)

**Comment by Whiteknight2119** [comment]
Experiment Overview:

For my experiment, I collected soil samples at two points (May and August 2023) from two locations, each containing two fields (No-Till and Conventional Tillage), totaling four fields. I processed these samples in the lab to investigate the effects of moisture variability on nitrogen in the rhizosphere under contrasting tillage systems. I applied two moisture treatments (Constant Moisture and Wet/Dry) and three exudate treatments (Oxalic Acid, Glucose, and H2O as control).

Experimental Design:

The soil samples from the four fields were subsampled into 20g of dry soil in a 3-way factorial design with three replicates. The full experimental design was replicated three times. The experimental units were as follows:

Timepoint Zero (Pre-incubated):
2 Locations (L1-CT, L1-NT, L2-CT, L2-NT)
2 Management types (CT, NT)
2 Moisture levels (CM, WD)
3 Exudate treatments (H2O, Glu, Ox)
3 Replicates
Total: 72 experimental units

Timepoint One (Main Incubation, 8 days):
2 Locations (L1-CT, L1-NT, L2-CT, L2-NT)
2 Management types (CT, NT)
3 Replicates
Total: 12 experimental units

Overall total: 84 experimental units

Additional Information:
I included an 'Experiment' factor to assess the correlation between May and August data. This is a separate section of the analysis.

**Comment by bbolker** [comment]
tl;dr **Please don't make us work so hard**.

Please see https://github.com/glmmTMB/glmmTMB/blob/master/README.md#where-to-ask-questions; most of your questions are better suited for one of the other venues (Stack Overflow, Cross Validated, or the r-sig-mixed mailing list)

Based on detailed comments below, I even more strongly suspect that you are having the problems described in [this comment of mine above](https://github.com/glmmTMB/glmmTMB/issues/1049#issuecomment-2174789514). Please follow those links, read them, and see if they resolve your problem.

---

I couldn't reproduce this. If possible you should give us a **minimal** reproducible example; i.e., condense all your data into a single, pre-processed CSV file that can be read directly into R with minimal further processing.  When I tried to work through your code I ran into

```r
Glm_MBN_Day0 <- glmer(MBN ~ Mgt + (1 | Loc),
   data = Day0, family = gaussian(link = "logit"))
> Error in family$linkfun(rho$mustart) : Value 17.68 out of range (0, 1)
```
... which is not surprising, as a logit link is intended for data on the range [0,1].

I added `knitr::opts_chunk$set(error = TRUE)` to allow the code to run past errors.

This model works fine for me:

```r
Gl_MBN_T1_Day8 <- glmmTMB::glmmTMB(MBN ~ Mgt*Moist_Trt*Ex_Trt + (1 | Loc),
                 data = Exp2_Data, family = gaussian)
```

... which makes me even more strongly suspect that you are having the problems described in [this comment of mine above](https://github.com/glmmTMB/glmmTMB/issues/1049#issuecomment-2174789514). Please follow those links, read them, and see if they resolve your problem.

When I run your last model,

```r
Gl_MBN_Day8_Combined <- glmmTMB(MBN ~ Mgt*Moist_Trt*Ex_Trt*Experiment + (1 | Loc),
                                data = Exp2_Data, family = gaussian)
```
 I get 

> Error in `contrasts<-`(`*tmp*`, value = contr.funs[1 + isOF[nn]]): contrasts can be applied only to factors with 2 or more levels

Looking more closely, `Exp2_Data` only has data from experiment 2, so we can't fit `Experiment` as a predictor. (More generally, considering just the three-way interaction gives us 12 parameters to estimate, which is too many for the 72 observations which are left once we drop incomplete cases ...
